src/main/resources/com/linecorp/armeria/server/docs/assets/js/armeria.js

```diff
@@ -183,34 +183,56 @@ $(function () {
     var toggleCollapser = function() {
       collapser.toggleClass('opened');
       collapser.next().collapse('toggle');
-    }
+    };
     collapser.click(function() {
       toggleCollapser();
     });
     var submitDebugRequest = function () {
-      var args;
+      var argsText;
+      var httpHeadersText = '';
       var httpHeaders = {};
+
+      // Validate arguments.
+      try {
+        var args = JSON.parse(debugText.val()); // Use the JSON parser for validation.
+        if (typeof args !== 'object') {
+          debugResponse.text("Arguments must be a JSON object.\nYou entered: " + typeof args);
+        }
+        // Do not use the parsed JSON but just a minified one not to lose number precision.
+        // See: https://github.com/line/armeria/issues/273
+        argsText = JSON.minify(debugText.val());
+      } catch (e) {
+        debugResponse.text("Failed to parse a JSON object:\n" + e);
+        return false;
+      }
+
+      // Validate HTTP headers.
       try {
         if (debugHttpHeadersText.val()) {
           httpHeaders = JSON.parse(debugHttpHeadersText.val());
+          if (typeof httpHeaders !== 'object') {
+            debugResponse.text("HTTP headers must be a JSON object.\nYou entered: " + typeof httpHeaders);
+            return false;
+          }
+
+          httpHeadersText = JSON.minify(debugHttpHeadersText.val());
+          if (httpHeadersText === '{}') {
+            httpHeadersText = '';
+          }
         }
-        args = JSON.parse(debugText.val());
       } catch (e) {
-        debugResponse.text("Failed to parse a JSON object, please check your request:\n" + e);
+        debugResponse.text("Failed to parse a JSON object:\n" + e);
```

Comment:
- Changed to `Failed to parse a JSON object in the HTTP headers field:`
- Made a similar change to the parse error message for the arguments field